### DISCUSSION TITLE
Disable sorting of bibliographic entries for a specific use case

### DISCUSSION THREAD

**Discussion by awa5114** [jgm/pandoc#10352] [Q&A] [answered]
Is there a way to bypass ordering of entries in my bibliography based on the order they have been cited in the main text? As a follow up to [this](https://github.com/jgm/pandoc/discussions/10147) discussion I am working on a website which will have a structure as follows:

```
---
title: "Main page"
---

- [Topic1](topic1.html)
- [Topic2](topic2.html)
- ....
- [References](references.html)
```

There is one standalone bibliography called `references.html`, an `index.html` and a number of `topic*.html` pages. I am citing literature from the central bibliography and linking to it. I wrote the following simple filter to achieve this:

```lua
-- all links inside citation objects need to be prepended by 
-- the relevant page, probably references.html
function Cite(elem)
    for key,value in pairs(elem.content) do
        if (value.t == "Link") then
            value.target = "./references.html" .. value.target
        end
    end
    return elem
end

-- Suppress bibliography from current page, suppressing using 
-- the CLI suppress option bypasses bibliographic cross 
-- referencing in pandoc filter
function Div(elem)
    if (elem.identifier) then
        return {}
    end

    return elem
end
```

I then build the pages individually applying the filter, which incidentally also suppresses the bibliography:

```
pandoc --citeproc -L filter.lua  --from markdown --to html --csl ieee.csl --bibliography references.bib index.md
```

Then I build the bibliography using a dummy markdown file as follows:

```
pandoc -s --citeproc --from markdown --to html --bibliography references.bib --csl ieee.csl -o references.html blank.md
```

While the links in my index.md point to the right bibliographic elements, they are incorrectly numbered. As it turns out the ordering of the bibliograpic elements  in `index.md` corresponds to the order they were cited, whereas in my central `references.html` they are numbered based on the `references.bib` file.

In my use case, it doesn't make sense to order the bibliographic items in terms of the order of citation, since the ordering is more or less arbitrary for each topic page.

How can I intentionally disable the pandoc behavior which causes the ordering of bibliographic entries to follow the order in which they have been cited?

**Comment by badumont** [comment, marked as answer]
This is determined by the CSL file, so you have to use a copy of ieee.csl and modify the <bibliography> element therein. Here is the relevant information: https://docs.citationstyles.org/en/v1.0.2/specification.html#sorting

**Reply by awa5114** [reply]
That seems to have worked, though i cannot find an option to sort them just based on the order they appear in the references.bib file
